build-tools/packages/build-cli/src/commands/check/policy.ts

```diff
@@ -0,0 +1,243 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+/* eslint-disable unicorn/prefer-module */
+/* eslint-disable import/no-internal-modules */
+/* eslint-disable @typescript-eslint/no-require-imports */
+/* eslint-disable unicorn/import-style */
+/* eslint-disable camelcase */
+/* eslint-disable @typescript-eslint/promise-function-async */
+/* eslint-disable array-callback-return */
+/* eslint-disable @typescript-eslint/strict-boolean-expressions */
+
+import * as fs from "fs";
+import { EOL as newline } from "os";
+import * as child_process from "child_process";
+import * as path from "path";
+import { Flags } from "@oclif/core";
+import {
+    copyrightFileHeaderHandlers,
+    npmPackageContentsHandlers,
+    dockerfilePackageHandler,
+    fluidCaseHandler,
+    lockfilesHandlers,
+    assertShortCodeHandler,
+    Handler,
+} from "@fluidframework/build-tools";
+import { BaseCommand } from "../../base";
+
+const readStdin: () => Promise<string | undefined> = () => {
+    return new Promise((resolve) => {
+        const stdin = process.openStdin();
+        stdin.setEncoding("utf-8");
+
+        let data = "";
+        stdin.on("data", (chunk) => {
+            data += chunk;
+        });
+
+        stdin.on("end", () => {
+            resolve(data);
+        });
+
+        if (stdin.isTTY) {
+            resolve("");
+        }
+    });
+};
+
+export class CheckPolicy extends BaseCommand<typeof CheckPolicy.flags> {
+    static description =
+        "Checks that the dependencies between Fluid Framework packages are properly layered.";
+
+    static flags = {
+        resolve: Flags.boolean({
+            description: `Resolve errors if possible`,
+            required: false,
+            char: "h",
+        }),
+        handler: Flags.string({
+            description: `Filter handler names by <regex>`,
+            required: false,
+            char: "h",
```

Comment:
```suggestion
            char: "d",
```

Conflicts with `-h` for help.